packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/ContextLogger.kt

```diff
@@ -0,0 +1,82 @@
+package io.realm.kotlin.internal
+
+import io.realm.kotlin.log.LogLevel
+import io.realm.kotlin.log.RealmLog
+
+/**
+ * Internal logger class used to inject context aware information into log message
+ * before they are passed on to the global [RealmLog].
+ */
+public class ContextLogger(public val context: String? = null) {
+
+    // define a context specific "prefix" for a log message.
+    // Note, this is in addition to the TAG defined by the RealmLogger.
+    // This can be used to add further context to a message, e.g:
+    // `[RealmTag] [CoreContext] message`
+    private val contextPrefix: String = if (context.isNullOrBlank()) {
+        ""
+    } else {
+        "[$context] "
+    }
+
+    // TRACE
+    public fun trace(throwable: Throwable?) {
+        RealmLog.doLog(LogLevel.TRACE, throwable, null)
```

Comment:
I would probably have kept a private 
```
inline fun doLog(...) { 
    RealmLog.doLog(...)
}
``` 
instead of replicating the `RealmLog`-dispatch in every method, but 🤷